presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1350,6 +1358,31 @@ else if (criteria instanceof JoinOn) {
             return output;
         }
 
+        private boolean needsJoinWarning(LogicalBinaryExpression expression)
+        {
+            ArrayDeque<Expression> queue = new ArrayDeque<Expression>();
```

Comment:
Nit: make the lefthand side just a `Deque<Expression>`

Reply:
Just make it Queue<Expression> is sufficient.

Reply:
Fixed